src/ripple/app/tx/impl/SusPay.cpp

```diff
@@ -329,10 +372,20 @@ SusPayFinish::doApply()
             ctx_.view().info().parentCloseTime.time_since_epoch().count())
         return tecNO_PERMISSION;
 
-    // Same digest?
-    if ((*slep)[~sfDigest] && (! ctx_.tx[~sfMethod] ||
-            (ctx_.tx[~sfDigest] != (*slep)[~sfDigest])))
-        return tecNO_PERMISSION;
+    // Check against condition in the ledger entry:
+    auto const cond = (*slep)[~sfCondition];
+
+    // If a condition wasn't specified during creation, one
+    // shouldn't be included now.
+    if (!cond && ctx_.tx[~sfCondition])
+        return temMALFORMED;
+
+    // If a condition was specified during creation of the
+    // suspended payment, the identical condition must also
+    // be presented. We don't check the fulfillment since we
+    // did that in preflight.
+    if (cond && (cond != ctx_.tx[~sfCondition]))
```

Comment:
`Condition::operator==` checks the type, featureBitmask, maxFulfillmentLength, and fingerprint. That's more than we want for this check. We only want to check type and fingerprint here.